File: Projects/PythonProject/Operators_And_Control_Flow/Operators.py
class Animal:
    pass

class Dog(Animal):
    pass

def main():
    # Arithmetic Operators
    a, b = 10, 5
    print("Arithmetic Operators:")
    print("Addition:", a + b)
    print("Subtraction:", a - b)
    print("Multiplication:", a * b)
    print("Division:", a / b)
    print("Modulus:", a % b)
    print("-----------------------------------")

    # Unary Operators
    print("Unary Operators:")
    x = 5
    print("Post-increment (simulated):", x)
    x += 1
    print("Pre-increment:", x)
    x -= 1
    print("Post-decrement (simulated):", x)
    x -= 1
    print("Pre-decrement:", x)
    print("Negation:", -x)
    print("-----------------------------------")

    # Assignment Operators
    print("Assignment Operators:")
    y = 10
    y += 5
    y -= 3
    y *= 2
    y /= 4
    y %= 3
    print("Relational Operators:")
    print("Equal:", a == b)
    print("Not Equal:", a != b)
    print("Greater Than:", a > b)
    print("Less Than:", a < b)
    print("Greater or Equal:", a >= b)
    print("Less or Equal:", a <= b)
    print("-----------------------------------")

    # Logical Operators
    print("Logical Operators:")
    p, q = True, False
    print("AND:", p and q)
    print("OR:", p or q)
    print("NOT:", not p)
    print("-----------------------------------")

    # Ternary Operator
    print("Ternary Operator:")
    min_val = a if a < b else b
    print("Minimum value:", min_val)
    print("-----------------------------------")

    # Bitwise Operators
    print("Bitwise Operators:")
    bit1, bit2 = 5, 3
    print("AND:", bit1 & bit2)
    print("OR:", bit1 | bit2)
    print("XOR:", bit1 ^ bit2)
    print("NOT:", ~bit1)
    print("-----------------------------------")

    # Shift Operators
    print("Shift Operators:")
    print("Left Shift:", a << 2)
    print("Right Shift:", a >> 2)
    print("-----------------------------------")

    # isinstance Operator
    print("isinstance Operator:")
    animal = Dog()
    print("Is Dog an instance of Animal?", isinstance(animal, Animal))
    print("Is Dog an instance of Dog?", isinstance(animal, Dog))

if __name__ == "__main__":
    main()

File: Projects/JavaProject/src/com/Operators_And_Control_Flow/Operators.java
package com.Operators_And_Control_Flow;

public class Operators {
    public static void main(String[] args) {
        // Arithmetic Operators
        int a = 10, b = 5;
        System.out.println("Arithmetic Operators:");
        System.out.println("Addition: " + (a + b));
        System.out.println("Subtraction: " + (a - b));
        System.out.println("Multiplication: " + (a * b));
        System.out.println("Division: " + (a / b));
        System.out.println("Modulus: " + (a % b));
        System.out.println("-----------------------------------");

        // Unary Operators
        System.out.println("Unary Operators:");
        int x = 5;
        System.out.println("Post-increment: " + (x++));
        System.out.println("Pre-increment: " + (++x));
        System.out.println("Post-decrement: " + (x--));
        System.out.println("Pre-decrement: " + (--x));
        System.out.println("Negation: " + (-x));
        System.out.println("-----------------------------------");

        // Assignment Operators
        System.out.println("Assignment Operators:");
        int y = 10;
        y += 5;
        y -= 3;
        y *= 2;
        y /= 4;
        y %= 3;
        System.out.println("Final value of y: " + y);
        System.out.println("-----------------------------------");

        // Relational Operators
        System.out.println("Relational Operators:");
        System.out.println("Equal: " + (a == b));
        System.out.println("Not Equal: " + (a != b));
        System.out.println("Greater Than: " + (a > b));
        System.out.println("Less Than: " + (a < b));
        System.out.println("Greater or Equal: " + (a >= b));
        System.out.println("Less or Equal: " + (a <= b));
        System.out.println("-----------------------------------");

        // Logical Operators
        System.out.println("Logical Operators:");
        boolean p = true, q = false;
        System.out.println("AND: " + (p && q));
        System.out.println("OR: " + (p || q));
        System.out.println("NOT: " + (!p));
        System.out.println("-----------------------------------");

        // Ternary Operator
        System.out.println("Ternary Operator:");
        int min = (a < b) ? a : b;
        System.out.println("Minimum value: " + min);
        System.out.println("-----------------------------------");

        // Bitwise Operators
        System.out.println("Bitwise Operators:");
        int bit1 = 5, bit2 = 3;
        System.out.println("AND: " + (bit1 & bit2));
        System.out.println("OR: " + (bit1 | bit2));
        System.out.println("XOR: " + (bit1 ^ bit2));
        System.out.println("NOT: " + (~bit1));
        System.out.println("-----------------------------------");

        // Shift Operators
        System.out.println("Shift Operators:");
        System.out.println("Left Shift: " + (a << 2));
        System.out.println("Right Shift: " + (a >> 2));
        System.out.println("-----------------------------------");

        // instanceof Operator
        System.out.println("instanceof Operator:");
        Animal animal = new Dog();
        System.out.println("Is Dog an instance of Animal? " + (animal instanceof Animal));
        System.out.println("Is Dog an instance of Dog? " + (animal instanceof Dog));
    }
}

// Defining classes for instanceof operator example
class Animal {
}

class Dog extends Animal {
}

